src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenInParametersTests.cs

```diff
@@ -4386,5 +4386,417 @@ .locals init (Test V_0)
+            // Note: we do a constrained call directly on the field
+            verifier.VerifyIL("D.M", """
+{
+  // Code size       70 (0x46)
+  .maxstack  1
+  .locals init (System.Collections.Generic.IEnumerator<int> V_0)
+  IL_0000:  ldarg.0
+  IL_0001:  ldflda     "S D.field"
+  IL_0006:  constrained. "S"
+  IL_000c:  callvirt   "System.Collections.Generic.IEnumerator<int> System.Collections.Generic.IEnumerable<int>.GetEnumerator()"
+  IL_0011:  stloc.0
+  .try
+  {
+    IL_0012:  br.s       IL_001b
+    IL_0014:  ldloc.0
+    IL_0015:  callvirt   "int System.Collections.Generic.IEnumerator<int>.Current.get"
+    IL_001a:  pop
+    IL_001b:  ldloc.0
+    IL_001c:  callvirt   "bool System.Collections.IEnumerator.MoveNext()"
+    IL_0021:  brtrue.s   IL_0014
+    IL_0023:  leave.s    IL_002f
+  }
+  finally
+  {
+    IL_0025:  ldloc.0
+    IL_0026:  brfalse.s  IL_002e
+    IL_0028:  ldloc.0
+    IL_0029:  callvirt   "void System.IDisposable.Dispose()"
+    IL_002e:  endfinally
+  }
+  IL_002f:  ldarg.0
+  IL_0030:  ldflda     "S D.field"
+  IL_0035:  constrained. "S"
+  IL_003b:  callvirt   "string object.ToString()"
+  IL_0040:  call       "void System.Console.Write(string)"
+  IL_0045:  ret
+}
+""");
+        }
+
+        [Fact, WorkItem(66135, "https://github.com/dotnet/roslyn/issues/66135")]
+        public void InvokeStructToStringOverrideOnInParameter()
+        {
+            var text = @"
+using System;
+
+class C
+{
+    public static void Main()
+    {
+        S1 s = new S1();
+        Console.Write(M(in s));
+        Console.Write(M(in s));
+    }
+    static string M(in S1 s)
+    {
+        return s.ToString();
+    }
+}
+struct S1
+{
+    int i;
+    public override string ToString() => (i++).ToString();
+}
+";
+
+            var comp = CompileAndVerify(text, expectedOutput: "00");
+
+            comp.VerifyIL("C.M", """
+{
+  // Code size       21 (0x15)
+  .maxstack  1
+  .locals init (S1 V_0)
+  IL_0000:  ldarg.0
+  IL_0001:  ldobj      "S1"
+  IL_0006:  stloc.0
+  IL_0007:  ldloca.s   V_0
+  IL_0009:  constrained. "S1"
+  IL_000f:  callvirt   "string object.ToString()"
+  IL_0014:  ret
+}
+""");
+        }
+
+        [Fact, WorkItem(66135, "https://github.com/dotnet/roslyn/issues/66135")]
+        public void InvokeAddedStructToStringOverrideOnInParameter()
+        {
+            var libOrig_cs = """
+public struct S
+{
+    int i;
+    public void Report() { throw null; }
+}
+""";
+            var libOrig = CreateCompilation(libOrig_cs, assemblyName: "lib");
+
+            var libChanged_cs = """
+public struct S
+{
+    int i;
+    public override string ToString() => (i++).ToString();
+    public void Report() { System.Console.Write("RAN "); }
+}
+""";
+            var libChanged = CreateCompilation(libChanged_cs, assemblyName: "lib");
+
+            var libUser_cs = """
+public class C
+{
+    public static string M(in S s)
+    {
+        return s.ToString();
```

Comment:
>return s.ToString(); [](http://example.com/codeflow?start=8&length=20)

It would be good to test this scenario on a read-only field and confirm that mutations are preserved when the method is invoked in an instance constructor and not preserved in a regular method.